apps/files_sharing/src/utils/GeneratePassword.js

```diff
@@ -49,10 +50,12 @@ export default async function() {
 		}
 	}
 
-	// generate password of 10 length based on passwordSet
-	return Array(10).fill(0)
-		.reduce((prev, curr) => {
-			prev += passwordSet.charAt(Math.floor(Math.random() * passwordSet.length))
-			return prev
-		}, '')
+	const array = new Uint8Array(10)
+	const ratio = passwordSet.length / 255
+	self.crypto.getRandomValues(array)
+	let password = ''
+	for (let i = 0; i < array.length; i++) {
+		password += passwordSet.charAt(array[i] * ratio)
+	}
+	return password
```

Comment:
Maybe easier to understand:

```suggestion
return Array.from(self.crypto.getRandomValues(new Uint8Array(10)))
	.map(x => x % passwordSet.length)
	.map(x => passwordSet.charAt(x))
	.join('')
```

Reply:
Nitpick²: `.map(passwordSet.charAt)` :exploding_head:

Reply:
ah! I had tried it before without `Array.from`

but also the proposal is doing two passes on the array while mine only does one

let's leave it for now

Reply:
actually three passes as Array.from might do another copy :-D